src/shogun/distance/Distance.cpp

```diff
@@ -350,51 +344,32 @@ SGMatrix<T> CDistance::get_distance_matrix()
 		result=SG_MALLOC(T, total_num);
 
 	int32_t num_threads=parallel->get_num_threads();
-	if (num_threads < 2)
-	{
-		D_THREAD_PARAM<T> params;
-		params.distance=this;
-		params.result=result;
-		params.start=0;
-		params.end=m;
-		params.total_start=0;
-		params.total_end=total_num;
-		params.n=n;
-		params.m=m;
-		params.symmetric=symmetric;
-		params.verbose=true;
-		get_distance_matrix_helper<T>((void*) &params);
-	}
-	else
-	{
-		D_THREAD_PARAM<T>* params = SG_MALLOC(D_THREAD_PARAM<T>, num_threads);
-		int64_t step= total_num/num_threads;
 
-		int32_t t;
- 
-		num_threads--;
-
-		#pragma omp for
-		for (t=0; t<num_threads; t++)
-		{
-			params[t].distance = this;
-			params[t].result = result;
-			params[t].start = compute_row_start(t*step, n, symmetric);
-			params[t].end = compute_row_start((t+1)*step, n, symmetric);
-			params[t].total_start=t*step;
-			params[t].total_end=(t+1)*step;
-			params[t].n=n;
-			params[t].m=m;
-			params[t].symmetric=symmetric;
-			params[t].verbose=false;
-
-			CDistance::get_distance_matrix_helper<T>(&params[t]);
-		}
 
-		SG_FREE(params);
+	
+	D_THREAD_PARAM<T>* params = SG_MALLOC(D_THREAD_PARAM<T>, num_threads);
+	int64_t step= total_num/num_threads;
+		int32_t t;
 
+	num_threads--;
```

Comment:
@piyushgoel997 dont get me wrong here but you are sending code without even compiling? even long time developers of shogun would have a hard time adding new features like that, let alone a person who has never touched shogun's code.

here's a manual how to develop shogun in general: https://github.com/shogun-toolbox/docs/blob/master/DEVELOPING.md

and yes that doesn't work for windows as till 3 hours ago shogun was not supported on windows. but still... you will need linux do get started with shogun... there are many ways to get a linux running under windwos, please use google to find out how to do it.

you can develop things on windows as well but you'll need Visual Studio for that... and some other things, like cmake